pkg/snapstore/snapshot.go

```diff
@@ -134,7 +134,8 @@ func ParseSnapshot(snapPath string) (*brtypes.Snapshot, error) {
 			s.IsFinal = true
 		}
 	}
-	unixTime, err := strconv.ParseInt(timeWithSnapSuffix[0], 10, 64)
+	creationTimeAndChunkSuffix := strings.Split(timeWithSnapSuffix[0], "@")
```

Comment:
[Here](https://github.com/gardener/etcd-backup-restore/pull/697/files#diff-fdebef784a0c29069a2bcd6a93ca1602a51c9bc83ec50df693da585e52eed31cR74-R92), we're appending the `@chunk` suffix to address issue encountered with the emulator `fakeGCS`.

@anveshreddy18, I'm curious about the rationale behind searching for `@` to parse the creationTime. It seems more straightforward to utilize the entire suffix instead of just `@`. 
```go
creationTimeAndChunkSuffix := strings.Split(timeWithSnapSuffix[0], "@")
```
---

"@shreyas-s-rao, @ishan16696, what do you think about adding a `chunkSuffix` option to the snapstore interface, just like we do with `prefix`? This way, not just `gcssnapstore`, but all storage providers could use it if needed."

Reply:
Rather, I would ask why do we need to use `@chunk` at all? Can't we simply use a `/` at the end to indicate that it's a "folder"? Although, I'm not aware of how fake-gcs forms URIs of the objects within the folder - whether it would end up having two `/`s, like `a/b//c`. Maybe this is something that you can explore @anveshreddy18

Reply:
Thanks for the suggestion @shreyas-s-rao , but I'm not particular about any suffix, but we need some sort of suffix definitely, pls refer to this [issue](https://github.com/fsouza/fake-gcs-server/issues/1434) on fake-gcs repo that I created. 

So in the event of adding a suffix (it could be anything for now), the question by @seshachalam-yv was instead of using a pre-set value to differentiate the time and snapshot suffix with the `@` char, can we pass a suffix from the snapstore struct itself so that it can be used both while creating the snapshot with the new name & also while parsing it in the snapshot.go file. 

So for eg: we use some `x` as suffix, then both while creating it and parsing it, we should use the same whole suffix we wanted. Instead of using `@chunk` while creating and just `@` while parsing the snapshot name, he's suggesting that we use the whole word we used to identify the chunk while parsing also. And that whole word should come from a snapstore rather than having the word in individual files. Wdyt?

Reply:
> can we pass a suffix from the snapstore struct itself so that it can be used both while creating the snapshot with the new name & also while parsing it in the snapshot.go file.

Ideally yes, it has to be done from the snapstore itself. If you plan to add suffix to the chunk files, then I am not in favour of using a new delimiter like `@`, when we can use a standard delimiter like `.`, which we already use for suffixing certain snapshot information. So the suffix would simply be `.chunk.<part-number>` possibly.

So in `ParseSnapshot()` (line 128), `timeWithSnapSuffix` can now have more to the suffix, which is `.chunk.<part-number>`, which indicates that the snapshot is a chunk object with chunk-part-number too. You will need to handle this along with the existing handling of compression suffix and final suffix.

Of course, you will also have to now form the snapshot name too according to this logic, only if fake GCS is being used.

@anveshreddy18 @seshachalam-yv @ishan16696 thoughts?

Reply:
Updated in the latest commit. I'm getting the suffix from `brtypes` instead of hardcoding it. PTAL